### DISCUSSION TITLE
How do I use vault?

### DISCUSSION THREAD

**Discussion by Penguin98kStudio** [Eugeny/tabby#5553] [Q&A] [answered]
How do I use vault?
How does it save passwords?

**Comment by Eugeny** [comment, marked as answer]
Once you enable the vault, new SSH passwords are saved into it, and when adding a new SSH private key, you'll have the option to import it into the vault.

You'll also have the option of encrypting the whole app config.

**Reply by Keeyuu** [reply]
I still don't understand and don't see a button or a place to add a private key in the vault. When I select the local private key file in the ssh configuration, it is not saved in the vault

**Comment by lawweiliang** [comment]
@Penguin98kStudio 
Please close the discussion if the question was answered. 
It would be helpful to make the discussion section clean. 

Click `mark as answered` and `close the discussion`. 😄 

![image](https://user-images.githubusercontent.com/34936929/228098333-222ac3d4-4c39-4c26-a2a8-52906647e00e.png)

**Comment by WilliamStam** [comment]
this is pretty old but came up as the top result when searching for "tabby vault location". is the vault a file? can we back it up with onedrive (on windows) or is the web sync the only way?

**Reply by WilliamStam** [reply]
o found the answer. nvm! 


> They're stored in the OS keychain if you didn't enable Vault, and in the Vault, if you did.
> If you're using Vault, you can backup the credentials by backing up your complete config file.
